src/components/tag/tag.test.jsx

```diff
@@ -107,15 +107,35 @@ describe('Tag', () => {
     });
   });
 
-  describe('onClear API', () => {
-    it('calls the onClear handler when the icon is clicked', () => {
-      const onClearSpy = jest.fn();
-
-      render(<Tag {...requiredProps} onClear={onClearSpy} />);
+  describe('onRemove API', () => {
+    it('calls the onRemove handler when the icon is clicked', () => {
+      const ref = createRef();
+      const onRemoveSpy = jest.fn();
+      render(<Tag {...requiredProps} onRemove={onRemoveSpy} ref={ref} />);
+      userEvent.click(screen.getByRole('button'));
+      expect(onRemoveSpy).toHaveBeenCalledWith({ target: expect.anything() });
+    });
 
-      userEvent.click(screen.getAllByRole('gridcell')[1]);
+    it('calls the onRemove handler when enter key is pressed', () => {
```

Comment:
```suggestion
    it('calls the onRemove handler when enter key is pressed while the remove icon is focused', () => {
```

Below as well. Just so we don't confuse that action buttons should remove tag items when the main content is focused.

Reply:
I've been thinking about this and I think if we try to be overly descriptive then our tests will become unwieldy pretty fast since our UI components will be pretty complex with a11y and composition. 

I think the `it` description should be pretty succinct and then the test itself will setup the correct state. Otherwise, when we want to update behavior then we'll need to revise / reorganize the test descriptions -- which honestly never happens since we tend to look for failed tests or add new tests.